### DISCUSSION TITLE
Filter on right hand side of simple JOIN

### DISCUSSION THREAD

**Discussion by WaltzingPenguin** [supabase/supabase#3546] [Questions] [answered]
What is the correct way to represent this query using the Supabase JS api?
```sql
SELECT users.email, logins.created_at
FROM logins JOIN users ON logins.user_id = users.user_id
WHERE users.app_id = 'b8540912-e337-4f79-b6b8-344921a70b30'
ORDER BY logins.created_at DESC
```

The Supabase documentation is pretty sparse about filtering on related tables.  I tried
```js
supabase
    .from('logins')
    .select('users(email), created_at')
    .eq('users.app_id', app_id)
    .order('created_at', { ascending: false })
```
but that will return rows where `users.email` is `null` (field is not nullable).  It seems to be running something roughly equivalent to:
```sql
SELECT users.email, logins.created_at
FROM logins
LEFT JOIN (
  SELECT *
  FROM users
  WHERE users.app_id = 'b8540912-e337-4f79-b6b8-344921a70b30'
  ) as users
ON logins.user_id = users.user_id
ORDER BY logins.created_at DESC
```
which I do not want.

**Comment by soedirgo** [comment, marked as answer]
Right now this isn't possible and you need to do the filtering client-side: https://github.com/supabase/postgrest-js/issues/197. There has been work done on PostgREST to make this possible though.

Alternatively you can try creating a view instead.

**Reply by WaltzingPenguin** [reply]
Thank you.  Views do indeed look like they are the correct answer for my project.
